### PR TITLE
DOC: Add missing entries in `numpy.testing` documentation

### LINKED ISSUE #21397: DOC: Missing entry for `assert_no_warnings`
### Issue with current documentation:

Looking at [the `numpy.testing` documentation page](https://numpy.org/doc/stable/reference/routines.testing.html#module-numpy.testing), I cannot find any entry referring to the [assert_no_warnings](https://github.com/numpy/numpy/blob/fd646bd693400ce408566674f266407672248cad/numpy/testing/_private/utils.py#L1817-L1850) function. Is this the expected behaviour, or is the entry actually missing?

### Idea or request for content:

_No response_

### PR DESCRIPTION
**Pull request by Patol75**
Closes #21397

It appears that some `numpy.testing` functions are not included in the corresponding documentation page. In this PR, I have added the missing functions in `routines.testing.rst`. Which missing functions to add, in which documentation category to add them and the order within each category are by no means definitive choices.

### CODE REVIEW

**Review by melissawm — APPROVED** [review]
This looks good to me, thanks @Patol75 ! 

I do wonder about possible deprecation warnings, but I guess that will come in time.

File: doc/source/reference/routines.testing.rst [hunk 1]
```diff
@@ -53,13 +57,16 @@ Decorators
    dec.skipif
    dec.slow
    decorate_methods
+   raises
```
Comment (mattip) [inline doc/source/reference/routines.testing.rst:60]: The docstring for `raises` has a warning that it should only be used with `nose`. I think we shouldn't index it.
```suggestion
```

**Comment by mattip** [comment]
Thanks @Patol75

**Comment by InessaPawson** [comment]
Hi-five on merging your first pull request to NumPy, @Patol75! We hope you stick around! Your choices aren’t limited to programming – you can review pull requests, help us stay on top of new and old  issues, develop educational material, work on our website, add or improve graphic design, create marketing materials, translate website content, write grant proposals, and help with other fundraising initiatives. For more info, check out: https://numpy.org/contribute
Also, consider joining our mailing list. This is a great way to connect with other cool people in our community and be part of important conversations that affect the development of NumPy: https://mail.python.org/mailman/listinfo/numpy-discussion

**Pull request merged**